pkg/channel/distributed/common/kafka/sarama/sarama.go

```diff
@@ -407,42 +263,37 @@ func MergeSaramaSettings(config *sarama.Config, configMap *corev1.ConfigMap) err
 	}
 
 	// Return Success
-	return nil
+	return config, nil
 }
 
 // Load The Sarama & EventingKafka Configuration From The ConfigMap
-func LoadConfigFromMap(configMap *corev1.ConfigMap) (*sarama.Config, *commonconfig.EventingKafkaConfig, error) {
-
-	// Validate The ConfigMap Data
-	if configMap.Data == nil {
-		return nil, nil, fmt.Errorf("attempted to load configuration from empty configmap")
+// The Provided Context Must Have A Kubernetes Client Associated With It
+func LoadSettings(ctx context.Context) (*sarama.Config, *commonconfig.EventingKafkaConfig, error) {
+	if ctx == nil {
+		return nil, nil, fmt.Errorf("attempted to load settings from a nil context")
 	}
 
-	// Create A New Base Sarama Config To Start From
-	saramaConfig := NewSaramaConfig()
-
-	// Merge The Sarama Settings In The ConfigMap Into the New/Base Sarama Config
-	err := MergeSaramaSettings(saramaConfig, configMap)
+	configMaps := kubeclient.Get(ctx).CoreV1().ConfigMaps(system.Namespace())
+	configMap, err := configMaps.Get(ctx, commonconfig.SettingsConfigMapName, metav1.GetOptions{})
 	if err != nil {
```

Comment:
seems odd (non-standard) to do the configmap loading on two lines instead of just....
```
configMap, err := kubeclient.Get(ctx).CoreV1().ConfigMaps(system.Namespace()).Get(ctx, commonconfig.SettingsConfigMapName, metav1.GetOptions{})
```

Reply:
That's true; originally I was checking to see if the configMaps variable was nil and I'm not sure why I removed that check but left it as separate lines.  Most other places that call CoreV1().ConfigMaps don't check for nil either though, and after a quick glance it looks like buried in the k8s.io code it just calls newConfigMaps() which can't ever return nil.  Will recombine as suggested.

Reply:
👍